src/ko/toonkor/src/eu/kanade/tachiyomi/extension/ko/toonkor/Toonkor.kt

```diff
@@ -213,7 +216,21 @@ class Toonkor : ConfigurableSource, ParsedHttpSource() {
         screen.addPreference(baseUrlPref)
     }
 
-    private fun getPrefBaseUrl(): String = preferences.getString(BASE_URL_PREF, defaultBaseUrl)!!
+    private fun getPrefBaseUrl(): String {
+        val prefUrl = preferences.getString(BASE_URL_PREF, defaultBaseUrl) ?: defaultBaseUrl
+        if (prefUrl != "auto") return prefUrl
```

Comment:
I am not too sure on adding "auto" in the url field...

maybe a switch pref to use auto or manual links, in auto mode, it will fetch the url and in manual mode it will use the one from preference.
can hide the manual link entering when auto is is enabled through `setVisible`

Reply:
I tested with `setVisible`, but the UX was kinda bad. You have to reopen the preferences screen to input the override URL. Opted to clarify the override URL option instead.

I set it to be enabled by default as it will lessen the support burden. The setting is kept across versions, as opposed to the override URL.

Reply:
I would say make the baseUrl pref also not depend on version.
implementation like [this](https://github.com/keiyoushi/extensions-source/blob/main/src/zh/zerobyw/src/eu/kanade/tachiyomi/extension/zh/zerobyw/UpdateUrl.kt#L24)